pkgs/development/compilers/bambu/default.nix

```
@@ -0,0 +1,257 @@
+  clangFrontends;
+assert builtins.all
+  (gcc:
+  lib.assertMsg (builtins.elem (lib.versions.major gcc.version) supportedGccMajorVersions) "GCC version ${gcc.version} is not supported as a frontend by bambu. The following major GCC versions are supported: ${builtins.concatStringsSep " " supportedGccMajorVersions}")
+  gccFrontends;
+# Assert that there is only one compiler frontend for each major version of the compiler
+let
+  passedClangVersions = (builtins.map (llvmPackages: llvmPackages.clang.version) clangFrontends);
+  duplicateClangVersion = lib.lists.findFirst (clangVersion: (builtins.length (builtins.filter (otherClangVersion: (lib.versions.major clangVersion) == (lib.versions.major otherClangVersion)) passedClangVersions)) > 1) false passedClangVersions;
+  passedGccVersions = (builtins.map (gcc: gcc.version) gccFrontends);
+  duplicateGccVersion = lib.lists.findFirst (gccVersion: (builtins.length (builtins.filter (otherGccVersion: (lib.versions.major gccVersion) == (lib.versions.major otherGccVersion)) passedGccVersions)) > 1) false passedGccVersions;
+in
+assert lib.assertMsg (duplicateClangVersion == false) "Multiple clang frontends with the same major version (${lib.versions.major duplicateClangVersion}) are not supported by bambu. You passed packages with the following versions: ${builtins.concatStringsSep " " (builtins.filter (clangVersion: (lib.versions.major clangVersion) == (lib.versions.major duplicateClangVersion)) passedClangVersions)}";
+assert lib.assertMsg (duplicateGccVersion == false) "Multiple gcc frontends with the same major version (${lib.versions.major duplicateGccVersion}) are not supported by bambu. You passed packages with the following versions: ${builtins.concatStringsSep " " (builtins.filter (gccVersion: (lib.versions.major gccVersion) == (lib.versions.major duplicateGccVersion)) passedGccVersions)}";
+stdenv.mkDerivation
+{
+  pname = "bambu";
+  version = "unstable-2023-08-01";
+  enableParallelBuilding = true;
+
+  src = fetchFromGitHub {
+    owner = "ferrandi";
+    repo = "PandA-bambu";
+    rev = "1d24ce0f8fe86686dc2366062a8d9707b685b3e6";
+    hash = "sha256-yIe2sRxbfVNvK39Y9zPLKaHwnxiI58jhJkVN/j96Q0A=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    autoconf-archive
+    automake
+
+    libtool
+
+    bison
+    doxygen
+    flex
+    graphviz
+    verilog
+    verilator
+  ];
+
+
+  buildInputs = [
+    boost
+    boost.dev
+    buddy
+    libmpc
+    mpfr
+    mpfr.dev
+    libxml2
+    libxml2.dev
+    xz
+    xz.dev
+    mpfi
+    zlib
+    zlib.dev
+    icu
+    icu.dev
+    suitesparse
+    suitesparse.dev
+    glpk
+    gmp
+    gmp.dev
+    gmpxx
+    gmpxx.dev
+  ];
+
+  patchPhase = ''
+    # Patch all calls to bash
+    find . -type f -exec sed -i 's|/bin/bash|${bash}/bin/bash|g' {} +
+  '';
+
+  configurePhase = ''
+    make -f Makefile.init
```

```suggestion
  configurePhase = ''
    runHook preConfigure
  
    make -f Makefile.init
```

And also `runHook postConfigure` below.